lib/floe/workflow/states/map.rb

```diff
@@ -74,7 +72,7 @@ def finish(context)
         def run_nonblock!(context)
           start(context) unless context.state_started?
 
-          loop while step_nonblock!(context) == 0 && running?(context)
+          step_nonblock!(context) while running?(context)
```

Comment:
is this a universal change you want to make?
(not asking for the change, just trying to understand trajectory)

Reply:
Specifically for map this was now essentially doing the same running? check twice (ended? is just !running?) so it was more efficient to do this